assignments/prototypes.js

```diff
@@ -1,28 +1,49 @@
 /*
   Object oriented design is commonly used in video games.  For this part of the assignment you will be implementing several constructor functions with their correct inheritance hierarchy.
 
-  In this file you will be creating three constructor functions: GameObject, CharacterStats, Humanoid.  
+  In this file you will be creating three constructor functions: GameObject, CharacterStats, Humanoid.
 
   At the bottom of this file are 3 objects that all end up inheriting from Humanoid.  Use the objects at the bottom of the page to test your constructor functions.
-  
+
   Each constructor function has unique properties and methods that are defined in their block comments below:
 */
-  
+
 /*
   === GameObject ===
   * createdAt
   * name
   * dimensions (These represent the character's size in the video game)
   * destroy() // prototype method that returns: `${this.name} was removed from the game.`
 */
+  function GameObject(attrs) {
+    this.createdAt = attrs.createdAt,
+ this.dimensions = attrs.dimensions,
+this.destroy = function() {
+  return `${this.name} was removed from the game.`;
+}
+}
 
 /*
   === CharacterStats ===
+  //CharacterStats.prototype = Object.create(GameObject.prototype);
```

Comment:
The other advantage of prototypes is that we can 'inherit' them by copying them over to a new object, with the line above. We would need to place this after we make the constructor function, but before we add any additional prototype functions, as we are overwriting the entire `prototype` object.